### PR TITLE
Engimarket 3, The Third III: Fucking Dance Machine 

### PR DESCRIPTION
**Pull request by Robustin**
Fuck you oranges, jesus christ. Yes Antur the "null" was removed merge meeeeee.

🆑 Robustin
add: The Engineering Dance Machine, along with assorted effects/sounds.
/🆑

### CODE REVIEW

**Comment by Jalleo** [comment]
This is a atomotisation remake PR of a pre merge feature. If it gets closed its due to ignorance.

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -0,0 +1,514 @@
...
+		if(t.x < cen.x && t.y == cen.y)
+			var/obj/item/device/flashlight/spotlight/L = new /obj/item/device/flashlight/spotlight(t)
+			L.light_color = "green"
+			L.light_power = 30-(get_dist(src,L)*8)
+			L.range = 1+get_dist(src, L)
+			spotlights+=L
+			continue
+		if((t.x+1 == cen.x && t.y+1 == cen.y) || (t.x+2==cen.x && t.y+2 == cen.y))
+			var/obj/item/device/flashlight/spotlight/L = new /obj/item/device/flashlight/spotlight(t)
+			L.light_color = "sw"
```
Comment (AnturK) [inline code/game/machinery/dance_machine.dm]: What's with these "sw" light color's and below

**Review by AnturK — COMMENTED** [review]
Just one question

**Comment by Man-Shroom-** [comment]
u can do it fam i believe in you

engineers need loot minigames too #equality

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (Robustin) [inline code/game/machinery/dance_machine.dm]: There are only 4 colors that rotate around the disco ball. This designates the diagonals which are initially off and allows the color rotation to proceed.

**Comment by WhiteHusky** [comment]
Would it make sense to use TGUI for the thing?

File: code/modules/power/singularity/emitter.dm [hunk 2]
```diff
@@ -281,7 +281,7 @@
 			else
 				to_chat(user, "<span class='warning'>The controls can only be locked when \the [src] is online!</span>")
 		else
-			to_chat(user, "<span class='danger'>Access denied.</span>")
+			to_chat(user, "<span class='danger'> denied.</span>")
```
Comment (Core0verload) [inline code/modules/power/singularity/emitter.dm]: why?

Comment (Robustin) [inline code/modules/power/singularity/emitter.dm]: Leftover from the atomization woops

**Comment by JJRcop** [comment]
in case anyone's wondering, Paradox - PlatinumEDM

**Pull request merged**